packages/openid4vc-client/src/OpenId4VcClientService.ts

```diff
@@ -3,34 +3,60 @@ import type { EndpointMetadata, Jwt } from '@sphereon/openid4vci-client'
 
 import {
   getKeyFromVerificationMethod,
+  AriesFrameworkError,
+  Buffer,
+  DidsApi,
   inject,
+  injectable,
   InjectionSymbols,
   isJwtAlgorithm,
-  Logger,
-  DidsApi,
-  AriesFrameworkError,
-  injectable,
   JsonEncoder,
   JsonTransformer,
-  W3cCredentialService,
-  W3cVerifiableCredential,
   JwsService,
   jwtKeyAlgMapping,
+  Logger,
+  TypedArrayEncoder,
+  W3cCredentialService,
+  W3cVerifiableCredential,
 } from '@aries-framework/core'
 import {
   Alg,
   AuthzFlowType,
+  CodeChallengeMethod,
   CredentialRequestClientBuilder,
   OpenID4VCIClient,
   ProofOfPossessionBuilder,
 } from '@sphereon/openid4vci-client'
+import { randomStringForEntropy } from '@stablelib/random'
```

Comment:
We should probably add a getRandomBytes method to the wallet so we can replace this. It's fine for now thiugh

Reply:
I've tried that method that Indy provides (genrateRandomSeed or something), but that one didn't meet the requirements for this use case (I forgot what the exact issue was).